crates/esthri/src/ops/copy.rs

```diff
@@ -12,42 +12,36 @@
 
 use log_derive::logfn;
 
-use crate::errors::{Error, Result};
-use crate::rusoto::S3;
 use crate::{
-    download, download_with_transparent_decompression, upload, upload_compressed, S3PathParam,
+    download,
+    errors::{Error, Result},
+    opts::*,
+    rusoto::S3,
+    upload, S3PathParam,
 };
 
 #[logfn(err = "ERROR")]
 pub async fn copy<T>(
     s3: &T,
     source: S3PathParam,
     destination: S3PathParam,
-    transparent_compression: bool,
+    opts: AwsCopyOptParams,
 ) -> Result<()>
 where
     T: S3 + Sync + Send + Clone,
 {
     match source {
         S3PathParam::Bucket { bucket, key } => match destination {
             S3PathParam::Local { path } => {
-                if transparent_compression {
-                    download_with_transparent_decompression(s3, bucket, key, path).await
-                } else {
-                    download(s3, bucket, key, path).await
-                }
+                download(s3, bucket, key, path, EsthriGetOptParams::from(opts)).await
```

Comment:
nit: `opts.into()` is less verbose - any reason not to go with it?

Reply:
sure